calplus/tests/unit/test_client.py

```diff
@@ -67,6 +70,53 @@ def test_client_called_with_cloud_config(self):
             version='1.0.0',
             resource='network',
             provider='amazon',
-            cloud_config=fake_aws_host
+            cloud_config=fake_aws_auth_opts
         )
         self.assertIsNotNone(test_client.driver)
+
+    def test_client_called_realize_all_driver(self):
+        def dict_to_opts_list(dict):
+            """
+            Convert cloud config to opts list
+            :param dict:
+            :return:
+            """
+            result = []
+            keys = dict.keys()
+            for key in keys:
+                if isinstance(dict[key], str):
+                    result.append(
+                        cfg.StrOpt(key, default=dict[key])
+                    )
+                else:
+                    result.append(
+                        cfg.DictOpt(key, default=dict[key])
+                    )
+            return result
+
+        # Add new aws host
+        amazon_group2 = cfg.OptGroup('amazon2',
+                                     title='Amazon Hosts')
+        CONF.register_group(amazon_group2)
+        list_opts = dict_to_opts_list(fake_aws_auth_opts)
+        CONF.register_opts(list_opts, group=amazon_group2)
+
+        enable_drivers = cfg.ListOpt(
+            'enable_drivers',
+            default=[
+                'openstack',
+                'amazon',
+                amazon_group2.name
+            ]
```

Comment:
totally wrong, I want to see the test case where user configured all cloud providers in the .conf file and then CAL can realize them all.

Please use some method for faking configuration file rather than using directly CONF in the unit test.